### DISCUSSION TITLE
tenantConfig attribute in Tenant item DynamoDB: ¿Additional Values?

### DISCUSSION THREAD

**Discussion by omarstapia** [awslabs/sbt-aws#124] [Q&A] [answered]
Hi everyone, im using [saas-reference-architecture-ecs](https://github.com/aws-samples/saas-reference-architecture-ecs), I have a question about records inserted into DynamoDB when provisioning a tenant. In the tenant item attributes, there is one called tenantConfig that contains something like this:
{
"userPoolId": "us-east-1_XXXXXXXXX",
"appClientId": "XXXXXXXXXXXXXXXXX",
"apiGatewayUrl": "https://xxxxxxxxx.execute-api.us-east-1.amazonaws.com/prod/"
}
My questions are:
-What do I need to do to add additional values to tenantConfig?
-What places these values in the record and where are they taken from? I've been going through the project but haven't been successful in inserting additional values, for example:
{
"userPoolId": "us-east-1_XXXXXXXXX",
"appClientId": "XXXXXXXXXXXXXXXXX",
"apiGatewayUrl": "https://xxxxxxxxx.execute-api.us-east-1.amazonaws.com/prod/",
"AdditionalValue":"XXXXXXXXXX"
}
Any observations or help would be greatly appreciated. Thanks!

**Comment by suhussai** [comment, marked as answer]
Hi @omarstapia ,

To answer your questions:

>  -What do I need to do to add additional values to tenantConfig?

You add them to the `tenantConfig` env var in your provisioning script. For example, for the ECS reference architecture, here's where the `tenantConfig` variable is defined.
https://github.com/aws-samples/saas-reference-architecture-ecs/blob/ad79aa13c0df85ab421284bfebb1b5a17870cc74/scripts/provision-tenant.sh#L93

> -What places these values in the record and where are they taken from? I've been going through the project but haven't been successful in inserting additional values, for example:
{
"userPoolId": "us-east-1_XXXXXXXXX",
"appClientId": "XXXXXXXXXXXXXXXXX",
"apiGatewayUrl": "https://xxxxxxxxx.execute-api.us-east-1.amazonaws.com/prod/",
"AdditionalValue":"XXXXXXXXXX"
}

SBT takes care of that for you. All you need to do is indicate which env var you'd like to "export" out from your provisioning job and SBT will take that and funnel that back and store it in a dynamodb table for you. For the ECS reference architecture, here's where we tell SBT that we want the `tenantConfig` env var (along with a few others) to be exported out from the provisioning job and saved with the tenant's information.

https://github.com/aws-samples/saas-reference-architecture-ecs/blob/ad79aa13c0df85ab421284bfebb1b5a17870cc74/server/lib/bootstrap-template/core-appplane-stack.ts#L52-L58

At a high level, there is a step function that manages the provisioning job. It's responsible for extracting the env vars you indicate and including them in the outgoing event payload (i.e., `PROVISION_SUCCESS` or `PROVISION_FAILURE`). Lastly, there is a rule that listens for these kinds of events, and calls out to the tenant management API to update the tenant with the information provided in event payload using tenant id and the outputs that you indicate as part of the provisioning job.

Hope that helps!

**Reply by omarstapia** [reply]
thanks buddy !
